Sources/XiEditor/Search.swift

```diff
@@ -421,7 +424,19 @@ extension EditViewController {
             if let resultCount = statusQuery["matches"] as? Int {
                 (query?.searchField as? FindSearchField)?.resultCount = resultCount
             }
+
+            if status.first?["lines"] != nil && !(status.first?["lines"] is NSNull) {
+                query?.lines = statusQuery["lines"] as! [Int]
```

Comment:
just an observation, related to my earlier comment about optionals: ideally by this point in the function we have validated the input and can just access members safely.